task/task_smoke_test.go

```diff
@@ -19,19 +19,62 @@ import (
 	"terraform-provider-iterative/task/common"
 )
 
+type testTarget struct {
+	machine string
+	image   string
+	script  string
+}
+
+var testTargets = map[string]testTarget{
+	// default test target spins up small instances and runs lightweight scripts on them.
+	"default": testTarget{
+		machine: "m",
+		image:   "ubuntu",
+		script: `
+#!/bin/sh -e
+mkdir --parents cache output
+touch cache/file
+echo "$ENVIRONMENT_VARIABLE_DATA" | tee --append output/file
+sleep 60
+cat output/file
+`[1:],
+	},
+	"gpu": testTarget{
+		machine: "m+t4",
+		image:   "nvidia",
+		script: `
+#!/bin/sh -e
+nvidia-smi
+mkdir --parents cache output
+touch cache/file
+echo "$ENVIRONMENT_VARIABLE_DATA" | tee --append output/file
+sleep 60
+cat output/file
+`[1:],
+	},
+}
+
 // TestTaskSmoke runs smoke tests with specified infrastructure providers.
 // Cloud provider access credentials (provided as environment variables) are required.
 func TestTaskSmoke(t *testing.T) {
 	testName := os.Getenv("SMOKE_TEST_IDENTIFIER")
 	sweepOnly := os.Getenv("SMOKE_TEST_SWEEP") != ""
-
 	enableAWS := os.Getenv("SMOKE_TEST_ENABLE_AWS") != ""
 	enableAZ := os.Getenv("SMOKE_TEST_ENABLE_AZ") != ""
 	enableGCP := os.Getenv("SMOKE_TEST_ENABLE_GCP") != ""
 	enableK8S := os.Getenv("SMOKE_TEST_ENABLE_K8S") != ""
 
 	enableALL := !enableAWS && !enableAZ && !enableGCP && !enableK8S
 
+	testTargetName := os.Getenv("SMOKE_TEST_TARGET")
+	if testTargetName == "" {
+		testTargetName = "default"
+	}
+	target, ok := testTargets[testTargetName]
+	if !ok {
+		t.Fatalf("test target %q undefined", testTargetName)
+	}
```

Comment:
Nitpick: maybe it's enough to use `targetName` for that identifier? 💅🏼 
```suggestion
	targetName := os.Getenv("SMOKE_TEST_TARGET")
	if targetName == "" {
		targetName = "default"
	}
	target, ok := testTargets[targetName]
	if !ok {
		t.Fatalf("test target %q undefined", targetName)
	}
```